addons/web_tour/static/src/js/tour_service.js

```diff
@@ -140,18 +148,41 @@ return session.is_bound.then(function () {
         });
 
         // Now that the observer is configured, we have to start it when needed.
+        const observerOptions = {
+            attributes: true,
+            childList: true,
+            subtree: true,
+            attributeOldValue: true,
+            characterData: true,
+        };
+
         var start_service = (function () {
             return function (observe) {
                 return new Promise(function (resolve, reject) {
                     tour_manager._register_all(observe).then(function () {
                         if (observe) {
-                            observer.observe(document.body, {
-                                attributes: true,
-                                childList: true,
-                                subtree: true,
-                                attributeOldValue: true,
-                                characterData: true,
+                            observer.observe(document.body, observerOptions);
+
+                            // If an iframe is added during the tour, its DOM
+                            // mutations should also be observed to update the
+                            // tour manager.
+                            const containsOrIsIframe = node => {
+                                const isElement = node.nodeType === 1;
+                                return isElement && (node.querySelector('.o_iframe') || node.matches('.o_iframe'));
```

Comment:
It's faster to do `matches` than to `querySelector`, so do it first (especially since when we add iframes in the DOM, we probably add them without wrapping parents?). Actually I would also write `.classlist.contains('o_iframe')` instead of using `matches` (probably faster too and more readable IMO.